packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -267,14 +219,75 @@ export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<v
     }, waitingTimeSpan);
   }
 }
+
+export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<void, FxError>> {
+  ctx.logProvider?.info(
+    format(getStrings().solution.DeployArmTemplates.StartNotice, PluginDisplayName.Solution)
+  );
+  let result: Result<void, FxError>;
+  try {
+    result = await doDeployArmTemplates(ctx);
+  } catch (error) {
+    result = err(
+      returnSystemError(
+        error,
+        PluginDisplayName.Solution,
+        SolutionError.FailedToDeployArmTemplatesToAzure
+      )
+    );
+  }
+  await ProgressHelper.endDeployArmTemplatesProgress();
+  ctx.logProvider?.info(
+    format(getStrings().solution.DeployArmTemplates.EndNotice, PluginDisplayName.Solution)
+  );
+  return result;
+}
+
+async function getParameterJson(ctx: SolutionContext) {
+  const parameterDir = path.join(ctx.root, baseFolder, parameterFolder);
+  const parameterDefaultFilePath = path.join(parameterDir, parameterDefaultFileName);
+  let parameterJsonString;
+  let parameterFilePath = parameterDefaultFilePath;
+  try {
+    parameterJsonString = await getExpandedParameter(ctx, parameterFilePath);
+  } catch {
+    const parameterTemplateFilePath = path.join(parameterDir, parameterTemplateFileName);
+    ctx.logProvider?.warning(
```

Comment:
we do not expect the parameter.default.json file always exists, so treat this message as information. By the way, if file already exists and some unexpected error happened (like handlebars error), we should not try to use the template file to generate a new default file.

The integration code is rough, let's polish the details.

Reply:
Fixed.